File: backend-go/internal/domain/menu.go
package domain

import (
	"context"

	"github.com/google/uuid"
	"github.com/shopspring/decimal"
)

type MenuItem struct {
	ID          uuid.UUID
	CategoryID  uuid.UUID
	Name        string
	Description *string
	Price       decimal.Decimal
	IsAvailable bool
}

type MenuCategory struct {
	ID        uuid.UUID
	Name      string
	SortOrder int
}

type MenuItemRepository interface {
	GetByID(ctx context.Context, id uuid.UUID) (*MenuItem, error)
	ListByCategory(ctx context.Context, categoryID uuid.UUID) ([]*MenuItem, error)
	ListCategories(ctx context.Context) ([]*MenuCategory, error)
}

File: website/src/types/menu.ts
export interface MenuCategory {
  ID: string;
  Name: string;
  SortOrder: number;
}

export interface MenuItem {
  ID: string;
  CategoryID: string;
  Name: string;
  Description?: string;
  Price: string; // Decimal comes as string from backend usually
  ImageURL?: string;
  IsAvailable: boolean;
}

export interface CartItem {
  menuItem: MenuItem;
  quantity: number;
  notes?: string;
}
